R/pkg/inst/tests/testthat/test_sparkSQL.R

```diff
@@ -502,20 +502,21 @@ test_that("jsonRDD() on a RDD with json string", {
 })
 
 test_that("test tableNames and tables", {
+  previous_count <- length(tableNames())
   df <- read.json(jsonPath)
   createOrReplaceTempView(df, "table1")
-  expect_equal(length(tableNames()), 1)
+  expect_true(length(tableNames()) == 1 + previous_count)
```

Comment:
this is one of the existing weak point in SparkR tests, we really should start/end each test case in a clean state such that one doesn't affect another.

In this case, is there a table/view that is left behind from before?